webpack/InsightsHostDetailsTab/InsightsTab.js

```diff
@@ -1,66 +1,62 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { orderBy } from 'lodash';
 import { Grid, ListView } from 'patternfly-react';
-import { noop } from 'foremanReact/common/helpers';
 import { translate as __ } from 'foremanReact/common/I18n';
+import { useSelector, useDispatch } from 'react-redux';
+import * as actions from './InsightsTabActions';
+import { selectHits } from './InsightsTabSelectors';
 import ListItem from './components/ListItem';
 import './InsightsTab.scss';
 
-class InsightsHostDetailsTab extends React.Component {
-  componentDidMount() {
-    const { fetchHits, hostID } = this.props;
-    fetchHits(hostID);
-  }
+// I have no idea where this component is used. It ain't the host details Insights tab.
+const InsightsHostDetailsTab = ({ hostID }) => {
+  const dispatch = useDispatch();
+  const hits = useSelector(selectHits);
 
-  render() {
-    const { hits } = this.props;
+  useEffect(() => {
+    dispatch(actions.fetchHits(hostID));
+  }, [hostID, dispatch]);
 
-    if (!hits.length) {
-      return <h2>{__('No recommendations were found for this host!')}</h2>;
-    }
-    const hitsSorted = orderBy(hits, ['total_risk'], ['desc']);
-    const items = hitsSorted.map(
-      (
-        {
-          title,
-          total_risk: totalRisk,
-          results_url: resultsUrl,
-          solution_url: solutionUrl,
-        },
-        index
-      ) => (
-        <ListItem
-          key={index}
-          title={title}
-          totalRisk={totalRisk}
-          resultsUrl={resultsUrl}
-          solutionUrl={solutionUrl}
-        />
-      )
-    );
-    return (
-      <div id="host_details_insights_tab">
-        <Grid.Row>
-          <Grid.Col xs={12}>
-            <h2>{__('Recommendations')}</h2>
-            <ListView id="hits_list">{items}</ListView>
-          </Grid.Col>
-        </Grid.Row>
-      </div>
-    );
+  if (!hits.length) {
+    return <h2>{__('No recommendations were found for this host!')}</h2>;
   }
-}
 
-InsightsHostDetailsTab.propTypes = {
-  hostID: PropTypes.number.isRequired,
-  fetchHits: PropTypes.func,
-  hits: PropTypes.array,
+  const hitsSorted = orderBy(hits, ['total_risk'], ['desc']);
+  const items = hitsSorted.map(
+    (
+      {
+        title,
+        total_risk: totalRisk,
+        results_url: resultsUrl,
+        solution_url: solutionUrl,
+      },
+      index
+    ) => (
+      <ListItem
+        key={index}
```

Comment:
Do we want to use something more unique for the key? 
```suggestion
        key={resultsUrl}
```